cpp/src/arrow/datum.cc

```diff
@@ -279,4 +279,23 @@ void PrintTo(const Datum& datum, std::ostream* os) {
   }
 }
 
+std::string ToString(Datum::Kind kind) {
+  switch (kind) {
+    case Datum::NONE:
+      return "None";
+    case Datum::SCALAR:
+      return "Scalar";
+    case Datum::ARRAY:
+      return "Array";
+    case Datum::CHUNKED_ARRAY:
+      return "ChunkedArray";
+    case Datum::RECORD_BATCH:
+      return "RecordBatch";
+    case Datum::TABLE:
+      return "Table";
+    default:
+      return NULL;
```

Comment:
Let's return "(unknown)" or possibly `std::to_string(static_cast<int>(kind))`

Reply:
Or, rather, make sure this doesn't happen?
```suggestion
      DCHECK(false);
      return "";
```

Reply:
Great. 

Seeing the same pattern here. 

```c++
std::string ToString(TimeUnit::type unit) {
  switch (unit) {
    case TimeUnit::SECOND:
      return "s";
    case TimeUnit::MILLI:
      return "ms";
    case TimeUnit::MICRO:
      return "us";
    case TimeUnit::NANO:
      return "ns";
    default:
      DCHECK(false);
      return "";
  }
}
```